### DISCUSSION TITLE
Suggestion about internationalization / translation.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#117] [Ideas]
on another project (Universal Blood Pressure Manager - UPMP to quote it), I met the site https://hosted.weblate.org [](https://hosted.weblate.org) which facilitates the translation of tools.
I remember from the user side that everything is online; we have a progress report in % ...
on the other hand I do not know the constraints on the developer side.
a suggestion among others

**Comment by textbrowser** [comment]
Qt offers a translation framework because it aims for inclusion. I've embedded the translation files in the latest release of BQ. Translating in general is flawed because: a) translators need to oversee changes, b) translators are sparse, c) volunteers are sparse, d) translating is difficult,  and e) translating is not entirely valued.

**Comment by textbrowser** [comment]
I could translate BQ into a multitude of languages with the help of online tools. This is monumental work. I do not add meaning to my existence by doing this. :)

**Comment by meteos77** [comment]
I agree with you completely
I don't think your precious time as a developer is spent translating into a language other than your own. Your preferred language is C++ (at least from my point of view)
and if others want their native language ... a little work to make their software more user friendly

**Comment by meteos77** [comment]
I come back to explain the suggestion
the advantage of putting the translations online:

I am on a train for 2 hours, I can translate with the web service easily. 

a) translators have to supervise changes:
all contributors have a common vision of the progress concerning their respective language.

b) translators are rare, c) volunteers are rare:
not everybody installs qtcreator or qtLinguist
the .ts.qm is what for many people; in fact in my opinion the translators are far from the computer programming and thus to offer them a simple web service would make it possible to be freed from its constraints.

- work on the most up-to-date version. : 
if I work in my corner on the .TS file but it is not finished ... (it is the case at the moment) I don't send it to you but another Frenchman can do the same. we can work twice on the same fields and no translation is produced at 100%.

- emulation - motivation: if a language is more advanced, contributors of other languages will want to do the same (French is at 60% and Italian at 90% that will push French to contribute).

e) the translation is not fully valued:
Make an acknowledgement in the software for the contributors/translators.

When a release is approaching, you make a call to all translators to try to do 100% for the X.Y version.

**Comment by textbrowser** [comment]
The translation text files are accessible online in the repository. As they're text files, they can be manually edited after being created by Linguist. Linguist is the preferred mechanism, however. Acknowledgments have always been included in the release notes (https://textbrowser.github.io/biblioteq/). I'll include a new panel for contributions. Some of the information is lost so it will not be included. :)

**Comment by meteos77** [comment]
I saw in release notes that you will change your nickname for Santa Claus :-)

what is the deadline to give you the translations to be incorporated in the release?

**Reply by textbrowser** [reply]
It's a planned release. Like a wish. Doesn't mean it'll be granted. :)

**Comment by meteos77** [comment]
French version of the day for those who compile :-)
[biblioteq_fr_FR.ts.tar.gz](https://github.com/textbrowser/biblioteq/files/7589950/biblioteq_fr_FR.ts.tar.gz)

**Reply by textbrowser** [reply]
> French version of the day for those who compile :-) [biblioteq_fr_FR.ts.tar.gz](https://github.com/textbrowser/biblioteq/files/7589950/biblioteq_fr_FR.ts.tar.gz)

Updated the repository.

**Comment by meteos77** [comment]
I'm looking at the translation of the documentation, I'm stuck on the screenshot that I would like to redo in French.
the accuracy of the text is much better with your font  ... What is it ?
there were lines to separate the main functions in the previous version

![image1](https://user-images.githubusercontent.com/56431039/144631719-c7c241c3-c61d-4550-98b8-b819beb75122.png)

![image1-originale](https://user-images.githubusercontent.com/56431039/144631730-3cd7100b-16bc-4420-98eb-8532c79c890a.png)

 I would like to find the best rendering an idea ?

**Reply by textbrowser** [reply]
Which Linux distribution are you using?

Noto Sans : Regular : 10

That's my font.

**Reply by meteos77** [reply]
LinuxMint 20.2

with Noto Sans regular 10 :
![image1-noto-10](https://user-images.githubusercontent.com/56431039/144634598-75d64090-b7cf-4f26-9d3d-59294a5056de.png)

not the same result :-(

![image1-originale](https://user-images.githubusercontent.com/56431039/144634270-f1f93b6e-73d8-4c24-bbf3-2dd37fb7010a.png)

**Reply by meteos77** [reply]
my font : Noto Sans CJK TC : 10  Regular  - capture with gimp

![comparatif-noto-sans-10](https://user-images.githubusercontent.com/56431039/144636600-6cfc587e-3751-4c99-b656-d3327514d9b9.png)

**Reply by textbrowser** [reply]
Are you concerned with antialiasing?

**Comment by meteos77** [comment]
there are so many changes :-) today's versio
[2021-12-03_biblioteq_fr_FR.ts.zip](https://github.com/textbrowser/biblioteq/files/7650846/2021-12-03_biblioteq_fr_FR.ts.zip)

**Reply by textbrowser** [reply]
Committed.

**Comment by textbrowser** [comment]
Are museums in France geographically centralized or decentralized?

**Reply by meteos77** [reply]
I don't understand the question?

**Reply by textbrowser** [reply]
Suppose I'm in a little town on the outskirts of France. Do I have to travel a great distance to visit a museum? That is, does France suffer from rich clusters of museums which are inaccessible because of distance or are the museums dispersed?

**Reply by meteos77** [reply]
the biggest ones are grouped in Paris
I don't know if this will answer your question but the visitors per year
https://www.culture.gouv.fr/en/Thematiques/Musees/Les-musees-en-France/Les-musees-de-France/Les-musees-nationaux-le-reseau-des-musees-detenteurs-des-collections-nationales

**Reply by textbrowser** [reply]
https://en.wikipedia.org/wiki/List_of_museums_in_France

https://en.wikipedia.org/wiki/List_of_tourist_attractions_in_Paris#/media/File:Paris_printable_tourist_attractions_map.jpg

**Reply by textbrowser** [reply]
Is there a correlation between cultural development and knowledge and centralized museums?

**Reply by meteos77** [reply]
I am puzzled by your questions 
the translator leaves me perplexed
I don't know how to answer you because the question is not clear

**Reply by textbrowser** [reply]
That's fine. Never mind. :)

**Reply by meteos77** [reply]
in the wikipedia list, there are many small museums
but if a subject is linked to a territory, the museum will be installed in this territory
for example : prehistory ; drawings in the lascau caves 
is located in the area of the cave
https://www.lascaux.fr/en
the big collections are rather in the capital Paris

**Reply by meteos77** [reply]
the museum of the landing will be towards the places of the event
https://en.normandie-tourisme.fr/discover/history/d-day-and-the-battle-of-normandy/sites-and-attractions/

between the museum of the caves of lascau and the museum of the landing beaches it is 8h00 of road and 600km

between the castle of versailles and the louvres it's 30-40 minutes by train

**Reply by textbrowser** [reply]
Yes, that was my curiosity. What about Germany? Is this a common thing with every country? Are cultural centers centralized or decentralized?

**Reply by meteos77** [reply]
I think that Europe is built on their capital
Paris for France, Vienna for Austria, Prague for the Czech Republic ...Rome for Italy, Athens for Greece ...
Berlin for Germany (may be a special case with the after 39/45 - GDR/FRG)

some countries are very centralized like France where the region of Paris represents almost 20% of the population of the country.
on the other hand, Spain and Germany are countries with regional traditions where the autonomy of each region is more pronounced.

**Reply by textbrowser** [reply]
Astute observation. :) Would decentralization improve human existence?

**Reply by meteos77** [reply]
depending on the subject and the way of looking at it
if I go back to the museums, the prehistoric cave is surely better managed on site than from the capital at 500km
but the train that can bring the traveler from another country ...

**Reply by textbrowser** [reply]
That's a geographical museum, that is, it exists because of its location.

**Reply by meteos77** [reply]
we are a small country 672 051km2 
in your country the capital of the state concentrates all the museums?

**Comment by meteos77** [comment]
I would like the different translators to intervene on this fascinating subject

**Reply by textbrowser** [reply]
I think the US follows European centralization.

**Reply by textbrowser** [reply]
Rural people don't have access to museums because of many reasons. Some of those reasons are cultural, which is a strange reason, but a valid one. Other, resources. Great distances. Geography is the great divider.

**Reply by meteos77** [reply]
I thought you were more decentralized but does the word have the same meaning as here?
you are a continental country so time difference, law by states ...

the states are autonomous between them?
the states of Europe have some common laws but not in all the domains, when it comes to the organization of the states, it's each one that manages in its own way according to its history
Europe as an administrative entity has only 60 years of existence and not with all the states.

**Reply by textbrowser** [reply]
Centralized food production requires the food to be distributed which creates waste for large geographies. Decentralized food production allows autonomy. This is of course my own limited belief.

The idea of decentralization grows into cultural aspects. I think it's an interesting concept in computers which touches upon life.

https://www.science.org/doi/10.1126/sciadv.1500253

"From ants (54) to bees (46) to neurons in the brain (55), a variety of systems productively use decentralized decision mechanisms. Our general notion of decentralized decision-making assumes that no single agent has direct access to information across all of the choices or the ability to make, and communicate, a final decision. Although each agent does have the ability to make limited judgments and decisions, it is the system as a whole that must integrate these activities into a final choice and action. Our model of decentralized decision-making abstracts beyond any one of these systems and aims to provide a deeper understanding of how such mechanisms behave."

**Reply by textbrowser** [reply]
Life originates the idea, computers imitate it.

**Reply by meteos77** [reply]
I put you the history of the library Department
it's a way to bring culture close to the rural people
I live in a rather rural area but the country being rather small I can go in 4 or 5 hours in the capital thanks to the trains
[history-BDP.txt](https://github.com/textbrowser/biblioteq/files/7651607/history-BDP.txt)
.

**Reply by textbrowser** [reply]
That study reflects upon closed systems and not systems of systems. Perhaps decentralization is a form of decentralizing ideas and then combining them into a solution for a system. There's also the possibility of pure decentralization; the goal being autonomy and side goals being cooperation between groups.

**Reply by textbrowser** [reply]
Oh well I was curious to see if countries such as France consider these concepts when they create a museum. Like, I would look at a place and think that a museum should exist there because people are under-represented in that region. Similar to schools, bus routes, mail, etc.

**Reply by meteos77** [reply]
for 70 years in France it is rather the opposite phenomenon see : [Metropolis](https://en.wikipedia.org/wiki/Metropolis)

with the covid-19, the tendency it is slowed down but the urban centers attract the companies... which have the work... and thus the inhabitants and thus the services (airport, train museum hospitals...)

**Reply by meteos77** [reply]
if a school doesn't have the quota of students, they close it and don't try to attract new inhabitants to perpetuate it (national leader)

the mayor (local leader) does everything to attract new families with children (even selling land to build his house for $1)
land use planning is not optimal
I lived 45 years in the capital region and now I am in a desert department :-)

**Reply by meteos77** [reply]
before I used to go to the media library managed by professionals to borrow books now I am a volunteer in the rural library and I am looking for a software :-) to be able to lend books

**Reply by textbrowser** [reply]
This is America too with some exceptions, also probably because of the pandemic. Which returns to the centralization of society. Companies go to cities because people go to cities for culture.

"If you build it, they will come."

So, if you disperse society, it can flourish in more places.

**Reply by meteos77** [reply]
I don't think it will be good for many people to live in rural areas because it will lead to an artificialization of the soil and therefore bad for the planet.

bees and ants don't have concrete to build their homes

**Reply by textbrowser** [reply]
It's an interesting problem. Unsustainable metropolis or small rural towns.

**Reply by meteos77** [reply]
after each country has a different feel 

NewYork     6 865 inhabitants per km2
Paris         20 641 inhabitants per km2
Dubai     150 647 inhabitants per km2

But if we take the metropolis the tendency is reversed
NewYork : 1096 inhabitants per km2
Paris :         732 inhabitants per km2

my imagination of NewYork is the skyscrapers while in the figures it must be a city spread out
source : https://fr.wikipedia.org/wiki/Densit%C3%A9_urbaine

**Reply by meteos77** [reply]
paris advertising and new york advertising
![pub-NY](https://user-images.githubusercontent.com/56431039/144662200-024c69d4-5ae9-4f78-aef5-e4340ee39988.jpeg)
![pub-paris](https://user-images.githubusercontent.com/56431039/144662207-9e35ca16-c26f-49d2-8104-38c492bdb83e.jpeg)

**Comment by textbrowser** [comment]
We're looking at densities instead of the cause of them.

**Reply by meteos77** [reply]
your article on bees is very interesting 
If bees with their decentralized decisions were the best, why are they disappearing in such large numbers?
Can a better informed bee make a better decision for the whole colony?

not being a bee i don't have the answer :-)

**Comment by textbrowser** [comment]
Perhaps we're poisoning them.

**Reply by meteos77** [reply]
no doubt but their decisions should take into account this fact, the colony lives near such fields but many of our people die so we will move to a more liveable place

**Comment by textbrowser** [comment]
Are you suggesting that bees are capable of noticing that we're poisoning ourselves?

**Comment by meteos77** [comment]
traduction problem
I am not a bee :-) 
I don't know if when they look at a field they know if it's for humans but they dance to indicate where their food is with their fellow bees, if after consuming this food many die or don't come back to the hive, they should dance again to say beware this food source is not suitable for us bees.
their problem is probably that everything is rotten in the same corner

**Comment by meteos77** [comment]
for French speakers - the home page of the BiblioteQ site (a bit redesigned in my way)
the publication history notes are not translated. see English version.

pour les francophones - la page d'accueil du site de BiblioteQ (un peu remaniée à ma façon)
l'historique des notes de publications ne sont pas traduites voir version Anglaise.

[BiblioteQ-fr.html.zip](https://github.com/textbrowser/biblioteq/files/7655913/BiblioteQ-fr.html.zip)

**Reply by textbrowser** [reply]
Committed. :)
